test/core/gc/struct.wast

```diff
@@ -50,6 +43,28 @@
 )
 
 
+;; Field names
+
+(module
+  (type (struct (field $x i32)))
+  (type $t1 (struct (field i32) (field $x f32)))
+  (type $t2 (struct (field i32 i32) (field $x i64)))
+
+  (func (param (ref 0)) (result i32) (struct.get 0 $x (local.get 0)))
+  (func (param (ref $t1)) (result f32) (struct.get 1 $x (local.get 0)))
+  (func (param (ref $t2)) (result i64) (struct.get $t2 $x (local.get 0)))
```

Comment:
It would be good to have tests for all combinations of #names and raw numbers for the type index and the field index if we don't have that already. Did you decide to do anything to anticipate having instructions with field indices but no type indices?

Reply:
I added a simple test for each combination. (To be honest, we could probably use more tests for the various struct/array instructions in general.)

In terms of anticipation, no, what do you have in mind?

Reply:
I had one idea for future-proofing the syntax described in the second paragraph here: https://github.com/WebAssembly/gc/issues/361#issuecomment-1481597070

Reply:
Can you elaborate on that idea? I'm not really understanding why the syntax `struct.get t (field t i)` is more future-proof than `struct.get t i`. What does repeating the type twice buy us?

Reply:
In the future if we have some `struct.get_unannotated <fieldidx>` instruction that has no type annotation, and if the text syntax for `fieldidx` contains a type index along with the symbolic field name, then we can still unambiguously figure out how to resolve the symbolic field name to an index. For the current instructions, which already contain type indices, the extra type index in the fieldidx syntax is redundant, so we can define abbreviations to make the repetition unnecessary.

Reply:
Why couldn't we just use the text syntax `struct.get_unannotated tidx? $x` in that case? (Allowing omitting the type index only when the field index is numeric.)

Reply:
That would work, too! If you'd be happy with that, then it sounds good to me.